src/App.tsx

```diff
@@ -1,55 +1,104 @@
-import React from 'react';
+import {
+  ChangeEvent,
+  FC,
+  MouseEvent,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import cn from 'classnames';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropDownContent } from './components/DropDownContent/DropDownContent';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+const debounce = <T extends (...args: never[]) => void>(
+  func: T,
+  delay: number,
+) => {
+  let timerId: NodeJS.Timeout;
+
+  return (...args: Parameters<T>): void => {
+    clearTimeout(timerId);
+    timerId = setTimeout(func, delay, ...args);
+  };
+};
+
+export const App: FC = () => {
+  const [peoples, setPeoples] = useState<Person[]>([]);
+  const [query, setQuery] = useState('');
+  const [isSelectedQuery, setIsSelectedQuery] = useState(false);
+  const [selectedSlug, setSelectedSlug] = useState('');
+
+  useEffect(() => {
+    setPeoples(peopleFromServer);
+  }, []);
+
+  const queryWithDelay = useCallback(debounce(setIsSelectedQuery, 300), []);
+
+  const changeQuery = (event: ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setQuery(value);
+
+    setQuery(value);
+    if (!value) {
+      queryWithDelay(false);
+    } else {
+      queryWithDelay(true);
+    }
+  };
+
+  const onSelectPerson = (
+    event: MouseEvent,
+    person: Person,
+  ) => {
+    event.preventDefault();
+
+    setSelectedSlug(person.slug);
+    setQuery(person.name);
+    setIsSelectedQuery(false);
+  };
+
+  const visiblePeoples = peoples.filter(person => (
+    person.name.toLowerCase().trim().includes(query.toLowerCase())
+  ));
+
+  const getPeopleBySlug = (slug: string) => {
+    return peoples.find(person => person.slug === slug);
+  };
+
+  const selectedPerson = getPeopleBySlug(selectedSlug);
+  const { name, born, died } = selectedPerson ?? {};
```

Comment:
It looks like a crutch

```suggestion
  const { name, born, died } = selectedPerson ?? {};
```